src/main/java/com/dataguise/saas/service/S3LfaService.java

```diff
@@ -0,0 +1,207 @@
+
+        dgClusterInfo.setAgentInfo(dgAgentClusterInfoList);
+        dgClusterInfo.setCloudAgentInfo(dgCloudAgentInfoList);
+        return dgClusterInfo;
+    }
+
+    /**
+     * Method is used for editing the cluster Info using ID
+     * @param dgAgentClusterDTO
+     * @return
+     * @throws BadGatewayException
+     * @throws BadRequestException
+     */
+    public Integer editClusterInfo(DgAgentClusterDTO dgAgentClusterDTO, Integer id) throws BadGatewayException, BadRequestException{
+        Integer clusterId=null;
+        try {
+            DgClusterInfo dgClusterInfo = createClusterInfo(dgAgentClusterDTO, id);
+            clusterId = restDriver.editClusterInfo(dgClusterInfo,id);
+
+        } catch (DgDrvException e) {
+            logger.error("create cluster failed " + e);
+            throw new BadGatewayException(e.getMessage(), HttpStatus.BAD_GATEWAY);
+        }
+        return clusterId;
+    }
+
+    /**
+     * Method is used for test the cluster
+     * @param clusterId
+     * @return
+     * @throws BadRequestException
+     * @throws BadGatewayException
+     */
+    public Boolean testCluster(Integer clusterId) throws BadRequestException, BadGatewayException {
+        if (clusterId == null) {
+            throw new BadRequestException("Cluster" + ErrorConstants.ERROR_MISSING_PATH_VARIABLE.getErrorMessage(), HttpStatus.BAD_REQUEST);
+        }
+        boolean testConnectionStatus = false;
+
+            try {
+                List<DgClusterInfo> clusterInfoList;
+                DgClusterInfo clusterInfo;
+                clusterInfoList = restDriver.loadClustersInfo(clusterId);
+                clusterInfo=clusterInfoList.get(0);
+                Integer lfaAgentId=clusterInfo.getAgentInfo().get(0).getAgentId();
+                Integer s3CloudAgentId=clusterInfo.getCloudAgentInfo().get(0).getAgentId();
+
+                if(restDriver.testAgentStatus(lfaAgentId)&&restDriver.testAgentStatus(s3CloudAgentId)){
+
+                    testConnectionStatus=true;
+                }
+                else{
+                    testConnectionStatus=false;
+                }
+
+            } catch (DgDrvException e) {
+                throw new BadGatewayException(e.getMessage(), HttpStatus.BAD_GATEWAY);
+            }
+            return testConnectionStatus;
+
+        }
+
+    /**
+     * This method is used for deleting the cluster
+     * @param clusterId
+     * @return
+     * @throws BadRequestException
+     * @throws BadGatewayException
+     */
+        public Integer deleteCluster(Integer clusterId) throws BadRequestException, BadGatewayException {
+        if(clusterId ==null){
+            throw new BadRequestException("Cluster " + ErrorConstants.ERROR_MISSING_PATH_VARIABLE.getErrorMessage(), HttpStatus.BAD_REQUEST);
+        }
+            try {
+                restDriver.deleteClusterInfo(clusterId);
+            } catch (DgDrvException e) {
+                logger.error("Delete connection failed: " + e);
```

Comment:
**Reporter**: PMD    
    **Rule**: GuardLogStatement
    **Severity**: ERROR
    **File**: /work/src/main/java/com/dataguise/saas/service/S3LfaService.java L175
    
    Logger calls should be surrounded by log level guards.&#10;&#10;&#10;Best Practices https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#guardlogstatement
    
*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1798654350>*